nav2_map_server/src/map_3d/map_io_3d.cpp

```diff
@@ -0,0 +1,382 @@
+            "The pcd file is invalid please pass a file name with extension .pcd");
+  }
+  std::cout << "file written";
+
+  if (pcd_file_name[0] != '/') {
+    // dirname takes a mutable char *, so we copy into a vector
+    std::vector<char> fname_copy(yaml_filename.begin(), yaml_filename.end());
+    fname_copy.push_back('\0');
+    load_parameters.pcd_file_name =
+      std::string(dirname(fname_copy.data())) + '/' + pcd_file_name;
+  } else {
+    load_parameters.pcd_file_name = pcd_file_name;
+  }
+  std::cout << "setting the origin";
+
+  // Get view point as position and orientation
+  // view point will be loaded from pcd file while reading
+  // If not provided by YAML, if it provided by both YAML and PCD then
+  // a warning will be issued if they are not equal and PCD  will take precedence
+  try {
+    // Load origin form yaml_file
+    std::vector<double> pcd_origin = yaml_get_value<std::vector<double>>(doc, "origin");
+    if (pcd_origin.size() != 7) {
+      throw std::invalid_argument("view_point is wrong");
+    }
+
+    // Convert to Pose
+    // Position
+    load_parameters.origin.position.x = pcd_origin[0];
+    load_parameters.origin.position.y = pcd_origin[1];
+    load_parameters.origin.position.z = pcd_origin[2];
+
+    // Orientation
+    load_parameters.origin.orientation.w = pcd_origin[3];
+    load_parameters.origin.orientation.x = pcd_origin[4];
+    load_parameters.origin.orientation.y = pcd_origin[5];
+    load_parameters.origin.orientation.z = pcd_origin[6];
+
+    load_parameters.use_ext_origin = true;
+  } catch (YAML::Exception & e) {
+    load_parameters.use_ext_origin = false;
+    std::cout << "[WARNING] [map_io_3d]: Couldn't load view_point from yaml file: "
+      "If not provided, we will try to load from pcd file" << std::endl;
+  }
+
+  return load_parameters;
+}
+
+void loadMapFromFile(
+  const LoadParameters & load_parameters,
+  sensor_msgs::msg::PointCloud2 & map_msg,
+  geometry_msgs::msg::Pose & origin_msg)
+{
+  sensor_msgs::msg::PointCloud2 map;
+  geometry_msgs::msg::Pose map_origin;
+
+  std::cout << "[INFO] [map_io_3d]: Loading pcd_file: " <<
+    load_parameters.pcd_file_name << std::endl;
+
+  pcl::PCLPointCloud2::Ptr cloud(new pcl::PCLPointCloud2());
+
+  //  create a pcd reader for PointCloud2 data
+  pcl::PCDReader reader;
+
+  Eigen::Vector4f position;
+  Eigen::Quaternionf orientation;
+
+  // PCD version fixed to version 0.7
+  // will use sensor origin (added in PCD version 0.7)
+  int pcd_version = pcl::PCDReader::PCD_V7;
+
+  std::cout << "[DEBUG] [map_io_3d]: feeding pcd filename to reader" << std::endl;
+
+  if (reader.read(
+      load_parameters.pcd_file_name, *cloud,
+      position, orientation, pcd_version) == -1)              //* load the file
+  {
+    std::string error_msg{"Couldn't read "};
+    error_msg += load_parameters.pcd_file_name + "\n";
+    PCL_ERROR(error_msg.c_str());
+  }
+  std::cout << "[DEBUG] [map_io_3d]: pcd file is loaded" << std::endl;
```

Comment:
this will always print, anything as a `cout` isn't a debug message because its always going to print no matter what. Either change all of these couts to infos or remove ones that are unnecessary to a user

Reply:
In general, this file seems to have excessive logging, but off of the top of my head, I dont know how much logging the 2D version of this file has. If its consistent with that, you can keep the logging here. Else, make it more consistent with the level of 2D IO logging